cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerService.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.worker;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.SConfiguration;
+import io.cdap.cdap.common.discovery.ResolvingDiscoverable;
+import io.cdap.cdap.common.discovery.URIScheme;
+import io.cdap.cdap.common.http.CommonNettyHttpServiceBuilder;
+import io.cdap.cdap.common.security.HttpsEnabler;
+import io.cdap.http.NettyHttpService;
+import org.apache.twill.common.Cancellable;
+import org.apache.twill.discovery.DiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * TaskWorkerService that launches an HTTP server for receiving and handling {@link RunnableTask}
+ */
+public class TaskWorkerService extends AbstractIdleService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TaskWorkerService.class);
+
+  private final CConfiguration cConf;
+  private final SConfiguration sConf;
+  private final DiscoveryService discoveryService;
+  private NettyHttpService httpService;
```

Comment:
You can have this final and construct it in the constructor.

Reply:
Done!